src/shims/unix/fs.rs

```diff
@@ -1660,6 +1689,36 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             }
         }
     }
+
+    fn isatty(&mut self, miri_fd: &OpTy<'tcx, Tag>) -> InterpResult<'tcx, i32> {
+        let this = self.eval_context_mut();
+        #[cfg(unix)]
+        {
+            let miri_fd = this.read_scalar(miri_fd)?.to_i32()?;
+            if let Some(host_fd) =
+                this.machine.file_handler.handles.get(&miri_fd).and_then(|fd| fd.as_unix_host_fd())
+            {
+                // "returns 1 if fd is an open file descriptor referring to a terminal;
+                // otherwise 0 is returned, and errno is set to indicate the error"
+                let is_tty = unsafe { libc::isatty(host_fd) };
```

Comment:
Could you add a SAFETY comment for this unsafe block?

Reply:
Yes? But as far as I know this function has no preconditions. It's an unusual one.

Reply:
"This function has no preconditions (and could be safe)" is a very useful SAFETY comment IMO.